router/usecase/router_usecase.go

```diff
@@ -262,30 +261,38 @@ func filterDuplicatePoolIDRoutes(rankedRoutes []route.RouteImpl) []route.RouteIm
 		}
 
 		// Add route to filtered ranked routes
-		filteredRankedRoutes = append(filteredRankedRoutes, route)
+		filteredRankedRoutes = append(filteredRankedRoutes, route.RouteImpl)
 	}
 	return filteredRankedRoutes
 }
 
 // rankRoutesByDirectQuote ranks the given candidate routes by estimating direct quotes over each route.
+// Additionally, it fileters out routes with duplicate pool IDs and cuts them for splits
+// based on the value of maxSplitRoutes.
 // Returns the top quote as well as the ranked routes in decrease order of amount out.
 // Returns error if:
 // - fails to read taker fees
 // - fails to convert candidate routes to routes
 // - fails to estimate direct quotes
-func (r *routerUseCaseImpl) rankRoutesByDirectQuote(ctx context.Context, candidateRoutes sqsdomain.CandidateRoutes, tokenIn sdk.Coin, tokenOutDenom string, maxRoutes int) (domain.Quote, []route.RouteImpl, error) {
+func (r *routerUseCaseImpl) rankRoutesByDirectQuote(ctx context.Context, candidateRoutes sqsdomain.CandidateRoutes, tokenIn sdk.Coin, tokenOutDenom string, maxSplitRoutes int) (domain.Quote, []route.RouteImpl, error) {
 	// Note that retrieving pools and taker fees is done in separate transactions.
 	// This is fine because taker fees don't change often.
 	routes, err := r.poolsUsecase.GetRoutesFromCandidates(candidateRoutes, tokenIn.Denom, tokenOutDenom)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	topQuote, routes, err := estimateDirectQuote(ctx, routes, tokenIn, maxRoutes, r.logger)
+	topQuote, routesWithAmtOut, err := estimateAndRankSingleRouteQuote(ctx, routes, tokenIn, r.logger)
 	if err != nil {
 		return nil, nil, fmt.Errorf("%s, tokenOutDenom (%s)", err, tokenOutDenom)
 	}
 
+	// Update ranked routes with filtered ranked routes
+	routes = filterAndConvertDuplicatePoolIDRankedRoutes(routesWithAmtOut)
+
+	// Cut routes for splits
+	routes = cutRoutesForSplits(maxSplitRoutes, routes)
```

Comment:
Note to reviewer: core change

1. Filter duplicate routes
2. Cut for splits

As opposed to the opposite order before causing problems